src/psim/truth/attitude_orbit.cpp

```diff
@@ -0,0 +1,155 @@
+  : Super(config, prefix, satellite, "eci") { }
+
+
+struct IntegratorData{
+  Real const &mass;
+  Vector3 const &I;
+  Real const &I_w;
+  Vector3 const &tau_w;
+  Vector3 const &m;
+  Vector3 const &b;
+};
+
+void AttitudeOrbitNoFuelGnc::step() {
+  this->Super::step();
+
+  // Initialize variables from .yml
+  auto &mass = prefix_satellite_m.get();
+  auto &I = prefix_satellite_J.get();
+  auto &I_w = prefix_satellite_wheels_J.get();
+  auto &tau_w = prefix_satellite_wheels_t.get();
+  auto &omega_w = prefix_satellite_wheels_w.get();
+  auto &m = prefix_satellite_magnetorquers_m.get();
+  auto &b = prefix_satellite_environment_b_eci->get();
+
+  auto &q_body_eci = prefix_satellite_attitude_q_body_eci.get();
+  auto &w = prefix_satellite_attitude_w.get();
+
+  // References to the current time and timestep
+  auto const &dt = prefix_dt_s->get();
+  auto const &t = prefix_t_s->get();
+
+  // References to position and velocity
+  auto &r = prefix_satellite_orbit_r.get();
+  auto &v = prefix_satellite_orbit_v.get();
+
+  IntegratorData data {mass, I, I_w, tau_w, m, b};
+
+  // Simulate our dynamics
+  
+  lin::Vector<Real, 16> const x = {
+      r(0), r(1), r(2),
+      v(0), v(1), v(2),
+      q_body_eci(0), q_body_eci(1), q_body_eci(2), q_body_eci(3),
+      w(0), w(1), w(2),
+      omega_w(0), omega_w(1), omega_w(2)
+  };
+
+  auto const xf = ode(t, dt, x, &data,
+      [](Real t, lin::Vector<Real, 16> const &x, void *ptr) -> lin::Vector<Real, 16> { // include/gnc/ode4.hpp
+    // position, velocity, quat_body, ang velocity, wheel ang velocity
+    auto const r = lin::ref<3, 1>(x, 0,  0);
+    auto const v = lin::ref<3, 1>(x, 3,  0);
+    auto const q = lin::ref<4, 1>(x, 6,  0);
+    auto const w = lin::ref<3, 1>(x, 10, 0);
+    auto const w_w = lin::ref<3, 1>(x, 14, 0);
+    auto *data = (IntegratorData *) ptr;
+
+    lin::Vector<Real, 16> dx;
+
+    // Orbital dynamics
+    {
+      // Calculate the Earth's current attitude (Position in ECI -> gravitational acceleration vector in ECI)
+      Vector4 q; 
+      gnc::env::earth_attitude(t, q);  // q = q_ecef_eci
+
+      // Determine our gravitation acceleration in ECI
+      Vector3 g;
+      {
+        Vector3 r_ecef;
+        gnc::utl::rotate_frame(q, r.eval(), r_ecef);
+
+        Real _;
+        gnc::env::gravity(r_ecef, g, _);  // g = g_ecef
+        gnc::utl::quat_conj(q);           // q = q_eci_ecef
+        gnc::utl::rotate_frame(q, g);     // g = g_eci
+      }
+
+      lin::ref<6, 1>(dx, 0, 0) = {v(0), v(1), v(2), g(0), g(1), g(2)};
+    }
+    
+    // dq = utl_quat_cross_mult(0.5*quat_rate,quat_body_eci);
+    Vector4 dq;
+    Vector4 quat_rate = {0.5 * w(0), 0.5 * w(1), 0.5 * w(2), 0.0};
+    gnc::utl::quat_cross_mult(quat_rate, q.eval(), dq);
+
+    // dw = w dot from equation
+    // Vector3 dw = inverse(data->I) * ((data->&m)).cross(data->&b) + (data->&tau_w) - (data->&omega).cross((data->&I))*(data->&omega)) + (data->&I_w)*(data->&omega_w))));
```

Comment:
Probs could shorten up this comment a bit:
```suggestion
    // dw/dt = I \ (cross(m, b) - tau_w - cross(w, I * w + I_w * w_w))
```